Sources/NIOOpenSSL/SSLPrivateKey.swift

```diff
@@ -80,6 +179,56 @@ public class OpenSSLPrivateKey {
         self.init(withReference: key!)
     }
 
+    /// Create an OpenSSLPrivateKey from a file at a given path in either PEM or
+    /// DER format, providing a passphrase callback.
+    ///
+    /// - parameters:
+    ///     - file: The path to the file to load.
+    ///     - format: The format of the key to load, either DER or PEM.
+    public convenience init(file: String, format: OpenSSLSerializationFormats) throws {
+        try self.init(file: file, format: format, pemUserData: nil)
+    }
+
+    /// Create an OpenSSLPrivateKey from a file at a given path in either PEM or
+    /// DER format, providing a passphrase callback.
+    ///
+    /// - parameters:
+    ///     - file: The path to the file to load.
+    ///     - format: The format of the key to load, either DER or PEM.
+    ///     - passphraseCallback: A callback to invoke to obtain the passphrase for
+    ///         encrypted keys.
+    public convenience init<T: Collection>(file: String, format: OpenSSLSerializationFormats, passphraseCallback: @escaping OpenSSLPassphraseCallback<T>) throws where T.Element == UInt8 {
+        let manager = OpenSSLPassphraseCallbackManager(userCallback: passphraseCallback)
+        let userData = Unmanaged.passRetained(manager).toOpaque()
```

Comment:
this should be matched with a `release` _in this function_ just like in the days before ARC. You don't pass a pointer that you own to someone else who needs to consume it as that's to error prone. The rules are that passing a pointer isn't passing ownership. If the consumer needs to own it for longer than its stack frame it needs to retain it.

Reply:
I'd actually probably do this:

```swift
withExtendedLifetime(manager) { manager in
    let managerPtr = Unmanaged.passUnretained(manager).toOpaque()
    ....(managerPtr)
}
```

Reply:
We can't actually do it that way, because the thing we have to put inside `withExtendedLifetime` is `self.init`, and if you try you get Swift complaining: "Initializer delegation ('self.init') cannot be nested in another expression"

Reply:
So I will have to manually manage the retains/releases. Not the end of the world, just a thing.